enclave_ops/ideep-enclave/Enclave/test.cpp

```diff
@@ -100,9 +102,16 @@ static sgx_status_t get_encryption_key(uint32_t model_id)
     }
     printf("Succeed to close Session...\n");
 
+    sgx_thread_mutex_lock(&global_mutex);
```

Comment:
Whether should the decryption process also be locked?

Reply:
seems global variants, e.g. g_model_keys, should be taken care of.

Reply:
The decryption process also uses the g_model_keys
ret = sgx_rijndael128GCM_decrypt((sgx_aes_gcm_128bit_key_t*)&g_model_keys[model_id],
                                     (uint8_t*)decrypt_handle, decrypt_bytes,
                                     (uint8_t*)decrypt_handle,
                                     (uint8_t*)iv_mac, gen_iv_size,
                                     NULL, 0,
                                     (sgx_aes_gcm_128bit_tag_t*)(iv_mac + gen_iv_size));
I'm not sure whether this would be rewrited when using

Reply:
g_model_keys here is read by not be written.

Reply:
Sure, I know it's read here.
Suppose following situation:  When a tensor is in decryption process,  whether there would be another thread is writing "g_model_keys" using "memcpy(&temp_key, &g_model_key, sizeof(temp_key));", the decryption did not lock this para and the decryption may goes wrong.
Could this happen?

Reply:
https://github.com/intel/pytorch/blob/sgx/enclave_ops/ideep-enclave/Enclave/test.cpp#L75
checked here. the key can only be added but cannot be updated.

Reply:
we can double check here as below:

  105  sgx_thread_mutex_lock(&global_mutex);
 +   /* model key had been retrieved, return directly. */
 +  if (g_model_keys.find(model_id) != g_model_keys.end())
 +       return SGX_SUCCESS;
    memcpy(&temp_key, &g_model_key, sizeof(temp_key));

Reply:
OK, that's good. Then I don't have other concerns.